### PR TITLE
Feature/cleanup

### PR DESCRIPTION
**Pull request by jbrnlz**
Some reorganisation of the main directory to avoid accumulation of files used on different systems or for different purposes:

- Compiliation, execution, and batch-submission files go into `templates/`, and can be copied from there when necessary. Git will not track any `*.sh` or `*.csh` files in the main directory. The updated `README.md` guides through this.

- Scripts for e.g. visualising or analysing output data (like in the `MATLAB` folder) go in `tools/`, anticipating the arrival of e.g. Python scripts for doing similar stuff.

- Folders related to benchmark or idealised experiments go in `benchmarks/`

Source code remains the same; default `Makefile` settings are now those needed for Snellius.

### CODE REVIEW

File: config-files/config_test [hunk 1]
```diff
@@ -3,7 +3,7 @@
   ! Time steps and range
   ! ====================
   
-  start_time_of_run_config                    = -100.0                           ! Start time (in years) of the simulations
+  start_time_of_run_config                    = -1.0                             ! Start time (in years) of the simulations
```
Comment (v1kko) [inline config-files/config_test:6]: isn't it better to leave this at -100? to allow all processes to happen at least once

File: src/utilities_module.f90 [hunk 2]
```diff
@@ -720,104 +720,104 @@ END SUBROUTINE line_integral_xydy
...
-    CALL allocate_shared_dp_2D( grid%ny + 2*n, grid%nx + 2*n, d_ext_smooth, wd_ext_smooth)
-    
+    CALL allocate_shared_dp_2D( grid%nx + 2*n, grid%ny + 2*n, d_ext,        wd_ext       )
+    CALL allocate_shared_dp_2D( grid%nx + 2*n, grid%ny + 2*n, d_ext_smooth, wd_ext_smooth)
+
     ! Copy data to the extended array and fill in the margins
     DO i = grid%i1, grid%i2
     DO j = 1, grid%ny
-      d_ext( j+n,i+n) = d( j,i)
+      d_ext( i+n,j+n) = d( i,j)
```
Comment (v1kko) [inline src/utilities_module.f90:754]: Fortran is column major, so the first rank should be iterated over in the inner loop for performance:

```fortran
   DO j = 1, grid%ny    
   DO i = grid%i1, grid%i2
       d_ext( i+n,j+n) = d( i,j)
  END DO
  END DO
```

File: src/utilities_module.f90 [hunk 3]
```diff
@@ -720,104 +720,104 @@ END SUBROUTINE line_integral_xydy
...
+    d_ext_smooth( grid%i1+n:grid%i2+n,:) = 0._dp
     CALL sync
-    
+
     DO i = grid%i1, grid%i2
     DO j = 1,       grid%ny
-      DO jj = -n, n
-        d_ext_smooth( j+n,i+n) = d_ext_smooth( j+n,i+n) + d_ext( j+n+jj,i+n) * f(jj)
+      DO ii = -n, n
+        d_ext_smooth( i+n,j+n) = d_ext_smooth( i+n,j+n) + d_ext( i+n+ii,j+n) * f(ii)
```
Comment (v1kko) [inline src/utilities_module.f90:781]: Same here, for performance the order of loops should be swapped

File: src/utilities_module.f90 [hunk 4]
```diff
@@ -720,104 +720,104 @@ END SUBROUTINE line_integral_xydy
...
     END DO
     END DO
     CALL sync
-    
+
     ! Copy data back
     DO i = grid%i1, grid%i2
     DO j = 1, grid%ny
-      d( j,i) = d_ext_smooth( j+n, i+n)
+      d( i,j) = d_ext_smooth( i+n, j+n)
```
Comment (v1kko) [inline src/utilities_module.f90:811]: Same here, do loops can be swapped

**Review by v1kko — CHANGES_REQUESTED** [review]
A few small suggestions (see comments). 
We could also add `-march=native` to (at least) the snellius compile flags to enable architecture dependent optimizations (or we can add it later).

Otherwise it compiles and runs fine for me on Snellius :)

File: config-files/config_test (hunk 1 above)
Comment (jbrnlz) [reply config-files/config_test:6]: Good point. I'll change it back once I finish debugging the mesh generation stuff.

File: src/utilities_module.f90 (hunk 2 above)
Comment (jbrnlz) [reply src/utilities_module.f90:754]: Thanks! : )

**Review by v1kko — APPROVED** [review]

**Pull request merged**